chore!: remove support for the deprecated assert/ syntax in configurable rules (#2089)
Redocly/redocly-cli: docs/@v2/guides/configure-rules.md

# Configure API linting

Every API has a different purpose and therefore different standards apply.
If you work in high security financial data, you may want a higher level of compliance with API standards than if you are creating a quick backend for a web application.

> Configuring an appropriate combination of rules is important; if the standards are too harsh, you risk ignoring errors and missing something important, but if they are too lax then your OpenAPI description may be inconsistent or incomplete for your needs.

Reviewer: ```suggestion
Configuring an appropriate combination of rules is important; if the standards are too strict, you risk ignoring errors and missing something important, but if they are too lax then your OpenAPI description may be inconsistent or incomplete for your needs.
```

  - recommended
```

The configuration here is _very_ minimal with only the ruleset defined so far.
Your config file also holds settings for themes, much more detailed configuration for linting and decorating, and custom plugins.

{% admonition type="info" %}
You can also define rulesets and other linting details differently for each API if you need to.
{% /admonition %}

## Adjust the rules in use

> The recommended ruleset aims to be best practice for all use cases, but of course every situation is different and some of the defaults may not fit your needs.

Reviewer: ```suggestion
The recommended ruleset aims to be best practice for all use cases, but every situation is different and some of the defaults may not fit your needs.
```

If you work with multiple APIs, or need to use consistent API linting in different projects, consider creating a ruleset that you can use in each situation.

Reusable rulesets are configured the same way as the Redocly CLI, but the yaml file contains only the `rules:` section.

> An example ruleset could be like this one, which is in a file called `reusable-ruleset.yaml`:

Reviewer: ```suggestion
Below you can see an example ruleset from a `reusable-ruleset.yaml` file:
```

## Use configurable rules

Assertions are a low-code way of creating targeted rules for specific situations in your use case.

> You specify which elements of the API description that should be checked, and using features like `defined`, `nonEmpty`, and `pattern`, describe the expectations for that element.

Reviewer: ```suggestion
Specify elements of the API description that you want to be checked, and using features like `defined`, `nonEmpty`, and `pattern`, describe the expectations for that element.
```

- No rules are defined at the top level, but since every API sets the `no-server-trailing-slash` rule to "off", this could be set at the top level.
- Each API adds (or removes) the rules that fit their use case, including using the `version-semver` assertion.

> Configuring per-API means that there doesn't have to be compromise for the lowest standard that all APIs can meet.

Reviewer: ```suggestion
Configuring per API means that  you don't have to compromise for the lowest standard that all APIs can meet.
```

> Especially when you are working on improving your APIs or API descriptions, setting the desired ruleset and adding exceptions until the API meets all requirements in full is a good way to ensure standards only improve.

Reviewer: ```suggestion
Especially when you are working on improving your APIs or API descriptions, setting the desired ruleset and adding exceptions until the API meets all requirements in full is a good way to ensure improvement of standards.
```
